### DISCUSSION TITLE
blocking "local" - why?

### DISCUSSION THREAD

**Discussion by hutchwilco** [AdguardTeam/AdGuardHome#2901] [Q&A]
AdGuard Home is operating with a handful of "standard" blocklists including Dan Pollock's List.
I have a lot of blocked requests that are simply labelled "local". I've no idea why you'd block this, or indeed why my various machines on my home network are simply making a DNS request for "local". It seems to happening often - several times a second. One machine is a server of sorts - Plex, Sonarr, Radarr etc, which all talk to each other so I'd expect them to be using local. 
Is Ad Guard Home nevertheless seeing this query, even though it's resolved within the network routing on the server itself?

Can anyone tell me why you'd bother blocking this?
![Screen Shot 2021-04-01 at 11 16 16 PM](https://user-images.githubusercontent.com/4792731/113280703-23efaf80-9341-11eb-9e38-e38013206070.png)

**Comment by ameshkov** [comment]
Probably, you should ask Dan Pollock about it, I don't understand why blocking it either

**Comment by MilesTEG1** [comment]
Hello,
@hutchwilco  Did you get a response from Dan Pollock?
I've got the same "local" blocked.

**Comment by Aikatsui** [comment]
@hutchwilco @MilesTEG1 Filter maintainers decisions and use cases sometimes may not fit for every users but you don't need to bother them. In this case, you can create a custom filtering rule to whitelist "local". as mentioned, it's enough to create one rule.

**Reply by MilesTEG1** [reply]
You're perfectly right.
I was just seeking for the answer of why some machines want to resolve the 127.0.0.1 IP witch is normally understood as localhost without entre needs of a dns server...
His list begins with this :
```Host

#<localhost>
127.0.0.1	localhost
127.0.0.1	localhost.localdomain
255.255.255.255	broadcasthost
::1		localhost
127.0.0.1	local
::1		ip6-localhost ip6-loopback
fe00::0		ip6-localnet
ff00::0		ip6-mcastprefix
ff02::1		ip6-allnodes
ff02::2		ip6-allrouters
ff02::3		ip6-allhosts
#fe80::1%lo0	localhost
#</localhost>
```
I'm not sure, but I checked my filter log and see this :
![image](https://user-images.githubusercontent.com/29298312/116171263-bd27a100-a708-11eb-861c-c870759161ca.jpeg)
My mba asked for a local domain.
I checked witch machine is asking for the local domaine : I just see my MBA and my iPhone.
My wife's iPhone isn't asking for such domain, nor my windows computer or any other machines (none Apple).
Do you know why some machines (probably only Apple) are asking for a local domaine and the others not ? (Maybe it's an application on both...)

**Reply by MilesTEG1** [reply]
If the blocklist is disabled, I got this from my iPhone :
![image](https://user-images.githubusercontent.com/29298312/116174108-10e8b900-a70e-11eb-816e-78d9a0e9bf7c.jpeg)
...
...
While I'm reading the comments lines of this blocklist I was understanding that this file was intended to be used on a computer...
That's why there is the bloc of `#<localhost>`.
But that's not answering why my machine aren't able to not let their requests to `local` or `localhost` hit the dns server...

It's a headache 😅

**Reply by DandelionSprout** [reply]
From what I know, in many hosts files, especially Windows' default system hosts file, `localhost` entries are used to tell Windows who the localhost is. So that Windows can look at the file and figure out *"Who was the localhost again? Oh, right, it's me, of course."* However, this does not seem to be needed for recent versions of Windows.

Such entries are sometimes (but rarely) used in third-party hosts files, in case end-users decide to just copypaste the entire list into the system hosts file.

**Comment by simonmcnair** [comment]
imo anything in localhost/hosts should not go to a dns server as they are local.  A request to 'localhost' cannot be resolved by a DNS server.  it is by definition the name of the local host and should be resolved to 127.0.0.1

**Reply by DandelionSprout** [reply]
In the years since the thread was created, new syntaxes have made it possible to use `|local|$dnsrewrite=127.0.0.1,dnstype=~AAAA|~CNAME`. I would however be pretty surprised if Team AdGuard implemented such an entry in AdGuard DNS Filter (or in similar hardcoded or opt-out functions in AdGuard Home) within the near future, unfortunately.

**Reply by iam-py-test** [reply]
[.local is not localhost; it's something on your LAN](https://en.wikipedia.org/wiki/.local)

**Reply by DandelionSprout** [reply]
While I figured from various research sources (Most of them my own semi-public DNS server) that anything with `(...).local` would be a very bad idea to block, it appear(s/ed) to me that lookups that are simply `local` with no periods or any other text at all would *presumably* be safe to redirect to 127.0.0.1?

**Reply by simonmcnair** [reply]
I would hope that many people realise the difference between using local as a host name and as a domain.  This conversation is as a host name if I understand properly.  You should not block .local but that is a lot different to local

**Reply by DandelionSprout** [reply]
As far as I can tell, these entries would be able to force resolvements to 127.0.0.1 indeed, as this thread's OP message indeed wants:
```
|local|$dnsrewrite=127.0.0.1,dnstype=~AAAA|~CNAME
|local|$dnsrewrite=::1,dnstype=AAAA
|localhost|$dnsrewrite=127.0.0.1,dnstype=~AAAA|~CNAME
|localhost|$dnsrewrite=::1,dnstype=AAAA
```
An entry that starts with a single `|` is that the request itself starts with that text, and is therefore extremely rarely used in major lists apart from some of mine (Other adblock tools would treat `http(...)` as the start of the requests), while ending with a single `|` is that the request ends with that text.
